eng/common/scripts/Prepare-Release.ps1

```diff
@@ -0,0 +1,132 @@
+#Requires -Version 6.0
+
+[CmdletBinding()]
+param(
+    [Parameter(Mandatory=$true)]
+    [string]$packageName,
+    [string]$serviceDirectoryName,
+    [string]$ReleaseDate,
+    [string]$BuildType # For Java
+)
+
+. ${PSScriptRoot}\common.ps1
+
+function Get-ReleaseDay($baseDate)
+{
+    # Find first friday
+    while ($baseDate.DayOfWeek -ne 5)
+    {
+        $baseDate = $baseDate.AddDays(1)
+    }
+    
+    # Go to Tuesday
+    $baseDate = $baseDate.AddDays(4)
+
+    return $baseDate;
+}
+
+$ErrorPreference = 'Stop'
+
+$packageProperties = Get-PkgProperties -PackageName $packageName -ServiceDirectory $serviceDirectoryName
+
+Write-Host "Source directory $serviceDirectoryName"
+
+if (Test-Path "Function:GetExistingPackageVersions")
+{
+    $existingVersions = GetExistingPackageVersions -PackageName $packageProperties.Name -GroupId $packageProperties.Group
+    if ($null -eq $existingVersions) { $existingVersions = @() }
+}
+else
+{
+    LogError "The function 'GetExistingPackageVersions' was not found."
+    return $null
+}
+
+if (!$ReleaseDate)
+{
+    $currentDate = Get-Date
+    $thisMonthReleaseDate = Get-ReleaseDay((Get-Date -Day 1));
+    $nextMonthReleaseDate = Get-ReleaseDay((Get-Date -Day 1).AddMonths(1));
+
+    if ($thisMonthReleaseDate -ge $currentDate)
+    {
+        # On track for this month release
+        $ParsedReleaseDate = $thisMonthReleaseDate
+    }
+    elseif ($currentDate.Day -lt 15)
+    {
+        # Catching up to this month release
+        $ParsedReleaseDate = $currentDate
+    }
+    else 
+    {
+        # Next month release
+        $ParsedReleaseDate = $nextMonthReleaseDate
+    }
+}
+else
+{
+    $ParsedReleaseDate = [datetime]::ParseExact($ReleaseDate, 'yyyy-MM-dd', [Globalization.CultureInfo]::InvariantCulture)
+}
+
+$releaseDateString = $ParsedReleaseDate.ToString("yyyy/MM/dd")
+$month = $ParsedReleaseDate.ToString("MMMM")
+
+Write-Host
+Write-Host "Assuming release is in $month with release date $releaseDateString" -ForegroundColor Green
+
+$latestVersion = $null;
```

Comment:
Lets remove the latestVersion stuff as I don't think it adds much value and does cause extra complication.